### PR TITLE
feat: DS custom pk updates for Async Storage Adapter and DB

### PR DESCRIPTION
**Pull request by david-mcafee**
#### Description of changes

- DS custom pk updates for Async Storage Adapter and DB
    - Refactor extraction of keys / values
    - Update how keys are stored in collection index map

#### Issue #, if available

N/A

#### Description of how you validated changes
- Manual tested with a sample app using 4 models to validate the usage of custom primary keys on all CRUD operations, including querying by PK (primary key), SK (sort key), and PK + SK.
- Updated unit tests

#### Checklist

- [ X ] PR description included
- [ X ] `yarn test` passes
- [ X ] Tests are [changed](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ X ] Relevant documentation is changed or added (and PR referenced)
    - Added in-line documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/datastore/src/storage/adapter/AsyncStorageDatabase.ts [hunk 1]
```diff
@@ -169,9 +174,17 @@ class AsyncStorageDatabase {
...
 
-			keysToDeleteArray.forEach(key =>
-				collection.delete(itemsMap[key].model[keys[0]])
-			);
+			keysToDeleteArray.forEach(key => {
+				// key: full db key
+				// keys: PK and/or SK keys
+
+				const primaryKeyValues: string = keys[0]
+					.split(DEFAULT_PRIMARY_KEY_SEPARATOR)
```
Comment (manueliglesias) [inline packages/datastore/src/storage/adapter/AsyncStorageDatabase.ts:182]: See the comment in the epic branch review about not using `.split()` over the keys, if you need back the parts of the key, build it using the info on the model definition instead

**Review by manueliglesias — COMMENTED** [review]
Left a comment regard `.split()` over the key

**Review by manueliglesias — APPROVED** [review]
LGTM

File: packages/datastore/src/storage/adapter/AsyncStorageAdapter.ts [hunk 2]
```diff
@@ -344,20 +346,17 @@ export class AsyncStorageAdapter implements Adapter {
 
 	private keyValueFromPredicate<T extends PersistentModel>(
 		predicates: PredicatesGroup<T>,
-		keyPathJoined: string[]
-	): string[] | undefined {
+		keys: string[]
```
Comment (manueliglesias) [inline packages/datastore/src/storage/adapter/AsyncStorageAdapter.ts:349]: This function returns false when the lengths are not equal

Do we need to fix the return type or make it return an empty array when key lengths don't match?

File: packages/datastore/src/storage/adapter/getDefaultAdapter/index.ts [hunk 3]
```diff
@@ -1,14 +1,14 @@
...
 
 const getDefaultAdapter: () => Adapter = () => {
 	const { isBrowser } = browserOrNode();
 
-	if ((isBrowser && window.indexedDB) || (isWebWorker() && self.indexedDB)) {
-		return IndexedDBAdapter;
-	}
+	// if ((isBrowser && window.indexedDB) || (isWebWorker() && self.indexedDB)) {
+	// 	return IndexedDBAdapter;
+	// }
```
Comment (manueliglesias) [inline packages/datastore/src/storage/adapter/getDefaultAdapter/index.ts:11]: Revert?

**Review by manueliglesias — APPROVED** [review]
LGTM, just a couple of comments to fix before merging

File: packages/datastore/src/storage/adapter/AsyncStorageAdapter.ts (hunk 2 above)
Comment (david-mcafee) [reply packages/datastore/src/storage/adapter/AsyncStorageAdapter.ts:349]: I'll update this to match the IndexedDB approach

**Review by iartemiev — APPROVED** [review]
LGTM 👍🏻

**Pull request merged**